### ISSUE TITLE
broom integration

### ISSUE THREAD

**Issue by acoppock** [DeclareDesign/estimatr#78]
is apparently BROKE.

MWE:

```
library(estimatr)
library(broom)

X <- rnorm(100)
Y <- rnorm(100)

fit <- lm_robust(Y ~ X)
tidy(fit) # errors
estimatr::tidy(fit)
```

**Comment by lukesonnet** [comment]
There's a decent explanation of this here: https://stackoverflow.com/a/18515795/4257918

Here's the problem:

When you load `broom` after `estimatr`, it clobbers the generic `tidy` method. Therefore, you are now calling `broom::tidy` whenever you call `tidy`. The problem is that both `broom::tidy` and `estimatr::tidy` are the following function:

```
tidy <- function(object, ...) UseMethod("tidy")
```
As per the `UseMethod` help docs:
```
To support this, UseMethod and NextMethod search for methods in two places: first in the environment in which the generic function is called, and then in the registration data base for the environment in which the generic is defined (typically a namespace).
```
So this means that when you call `broom::tidy`, `UseMethod` will only search the environment in which it is called and the `package:broom` namespace, because that is where `broom::tidy` is defined.

So there are two solutions:
1. Add a broom dependency
2. Forcefully export `tidy.lm_robust` (and the other methods) to the environment in which users load `estimatr`. This means when the user calls `broom::tidy` it is content to look in the current environment, to which we now forcefully export `tidy.lm_robust`. This doesn't solve the reverse problem, where if you load our package after theirs, then we have clobbered their default `tidy` method, and we can't force them to export all of their `tidy.lm` functions.

**Comment by lukesonnet** [comment]
Neal when you look at this, could you let me know what about what I said in the meeting was technically incorrect? I've just cobbled together what I could from man pages and Adv-R, but I couldn't find an easy resource to explain this to me.

**Comment by nfultz** [comment]
just to be explicit, we have three use cases -

```
require(estimatr)

model2 <- lm_robust(extra~group, sleep, clusters = "ID")

environment(tidy)
tidy(model2)


#vs

require(broom)
require(estimatr)

model <- lm(extra~group, sleep)
model2 <- lm_robust(extra~group, sleep, clusters = "ID")

environment(tidy)
tidy(model)
tidy(model2)

#vs

require(estimatr)
require(broom)

model <- lm(extra~group, sleep)
model2 <- lm_robust(extra~group, sleep, clusters = "ID")
environment(tidy)
tidy(model)
tidy(model2)

```

**Comment by nfultz** [comment]
See https://github.com/DeclareDesign/estimatr/tree/nfultz/s3_hell for a workaround

**Comment by lukesonnet** [comment]
Yes, but also all of our other tidy methods need to similarly be exported explicitly. Is there a way to do this through roxygen or must we manually add each of them each time.

**Comment by nfultz** [comment]
Updated the PR with the extra roxygen tag. There isn't really a use for us exporting tidy.data.frame and tidy.NULL for broom::tidy to use, it has it's own.

**Comment by lukesonnet** [comment]
Correct about the last two tidy fns.

**Comment by lukesonnet** [comment]
Closed by #82.
